### PR TITLE
Stripe link fixes

### PR DESCRIPTION
**Pull request by syvlabs**

### CODE REVIEW

File: lib/components/form.tsx [hunk 1]
```diff
@@ -483,11 +494,24 @@ const ElementsForm: FC<ElementsFormProps> = (props) => {
...
+    (cpm: CheckoutPaymentMethod) => onUserCompletePaymentRequestUI(null, cpm),
+    dynamicPreview
+  );
+
   const childrenProps: ElementsFormChildrenProps = {
     submit: submitCard,
     applePay: paymentRequests.apple_pay,
     googlePay: paymentRequests.google_pay,
-    stripeLink: paymentRequests.stripe_link,
+    stripeLink: {
```
Comment (zeyarpaing) [inline lib/components/form.tsx:510]: client need to render stripe link button and authElement or is it optional?

File: lib/components/stripe-link-button.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  className?: string;
+  // Note that this can be from 40-55 px only
+  buttonHeight?: number;
+};
+
+const OJS_STRIPE_LINK_ID = 'ojs-stripe-link-checkout';
+
+const mountStripeLinkButton = (buttonHeight?: number): void => {
+  if (buttonHeight !== undefined && buttonHeight > 55 && buttonHeight < 40) {
+    console.warn(`stripeLink.button height must be between 40 and 55px. Got: ${buttonHeight} px`);
```
Comment (zeyarpaing) [inline lib/components/stripe-link-button.tsx:14]: Stripe is crazy lol. should we provide default height?

File: lib/components/stripe-link-auth-element.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    // });
+    // expressCheckoutElement.mount(`#${OJS_STRIPE_LINK_ID}`);
+    // expressCheckoutElement.on('confirm', (event) => {
+    //   console.log('Stripe link confirmed', event);
+    //   confirmListener();
+    // });
+    // const linkAuthenticationElement = elements.create('linkAuthentication');
+    // linkAuthenticationElement.mount(`#${OJS_STRIPE_LINK_AUTH_ID}`);
+
+    const paymentElement = elements.create('payment', {
```
Comment (zeyarpaing) [inline lib/components/stripe-link-auth-element.tsx:31]: stripe's `payment` element is actually Card + other payment methods right? while linkAuth element is just an email field

File: lib/utils/stripe.ts [hunk 4]
```diff
@@ -31,12 +37,36 @@ const getLoadedStripe = async (publishableKey: string): Promise<StripeType> => {
...
+  return {
+    elements: stripe.elements(elementsOptions),
+    stripe,
+  };
+};
+
+export const createElementsOptions = (amount: Amount): StripeElementsOptionsClientSecret => {
+  console.log(amount);
+  return {
+    clientSecret: 'seti_1QCcXVKKXdhjXGwFhd0btSQD_secret_R4m53UhbceUDHVoxq07r2zoEgqJg7wd',
```
[the remark this answers is not in the document]
Comment (syvlabs) [reply lib/utils/stripe.ts:54]: Oh I forgot about this, thanks

File: lib/components/stripe-link-button.tsx (hunk 2 above)
Comment (syvlabs) [reply lib/components/stripe-link-button.tsx:14]: Yep default is `undefined`, which stripe sets to something decent

File: lib/utils/stripe.ts [hunk 5]
```diff
@@ -50,6 +80,7 @@ export const createStripePaymentRequest = async (
     },
     requestPayerName: true,
     requestPayerEmail: true,
+    disableWallets: !isLinkOnly ? [] : ['applePay', 'browserCard', 'googlePay'],
```
[the remark this answers is not in the document]
Comment (syvlabs) [reply lib/utils/stripe.ts:83]: No idea about what `browserCard` is, but these are all the possible values in that enum. The goal here is to only enable `link`

**Review by zeyarpaing — APPROVED** [review]

**Pull request merged**